File: MonoScripting/Api/EngineClass/EntityComponents/CharacterControllerComponent.cs
using System;
using System.Runtime.CompilerServices;
using System.Runtime.InteropServices;

public class CharacterControllerComponent : IEntityComponent {
    public CharacterControllerComponent(ulong handle) : base(handle)
    {
    }

    public CharacterControllerComponent() : base(0)
    {
    }

    public bool IsOnGround(){
        return i_IsOnGround(handle);
    }

    public void Move(Vec2 direction){
        i_Move(handle, direction);
    }
    
    public Vec3 GetCenter(){
        return i_GetCenter(handle);
    }

    public Vec2 GetSize(){
        return i_GetSize(handle);
    }

    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern bool i_IsOnGround(ulong handle);
   
    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern bool i_Move(ulong handle, Vec2 dir);

    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern Vec3 i_GetCenter(ulong handle);

    [MethodImpl(MethodImplOptions.InternalCall)]
    private static extern Vec2 i_GetSize(ulong handle);
}

File: Engine/MonoScripting/Api/EntityComponents/CharacterControllerComponent.cpp
#include "../ApiBindings.hpp"
#include <Scene/EntityComponents/CharacterControllerComponent.hpp>
#include <MonoScripting/MonoScriptingLayer.hpp>

using namespace VSGE;

bool IsOnGround(CharacterControllerComponent* ptr) {
	return ptr->IsOnGround();
}

void Move(CharacterControllerComponent* ptr, Vec2 direction) {
	ptr->Move(direction);
}

void VSGE::BindCharacterController() {
	MonoScriptingLayer::AddInternalCall("CharacterControllerComponent::i_IsOnGround(ulong)", IsOnGround);
	MonoScriptingLayer::AddInternalCall("CharacterControllerComponent::i_Move(ulong,Vec2)", Move);
}